pkg/proxy/nftables/proxier.go

```diff
@@ -1205,6 +1185,48 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 		}
 
+		usesFWChain := len(svcInfo.LoadBalancerVIPStrings()) > 0 && len(svcInfo.LoadBalancerSourceRanges()) > 0
+		fwChain := svcInfo.firewallChainName
+		if usesFWChain {
+			ensureChain(fwChain, tx, activeChains)
+			var sources []string
+			allowFromNode := false
+			for _, src := range svcInfo.LoadBalancerSourceRanges() {
+				_, cidr, _ := netutils.ParseCIDRSloppy(src)
+				if cidr == nil {
+					continue
+				}
+				if len(sources) > 0 {
+					sources = append(sources, ",")
+				}
+				sources = append(sources, src)
+				if cidr.Contains(proxier.nodeIP) {
+					allowFromNode = true
+				}
+			}
+			// For VIP-like LBs, the VIP is often added as a local
+			// address (via an IP route rule).  In that case, a request
+			// from a node to the VIP will not hit the loadbalancer but
+			// will loop back with the source IP set to the VIP.  We
+			// need the following rules to allow requests from this node.
+			if allowFromNode {
+				for _, lbip := range svcInfo.LoadBalancerVIPStrings() {
+					sources = append(sources, ",", lbip)
+				}
+			}
+			tx.Add(&knftables.Rule{
+				Chain: fwChain,
+				Rule: knftables.Concat(
+					ipX, "saddr", "{", sources, "}",
+					"return",
+				),
+			})
+			tx.Add(&knftables.Rule{
+				Chain: fwChain,
+				Rule:  "drop",
+			})
```

Comment:
hm... so actually, you could just add a `"!="` to the first rule and change `"return"` to `"drop"` and then not need the second rule, right? (And then flip around what I said about `helpers_test.go` before and only implement the "!=" version rather than only implementing the non-"!=" version.)

Reply:
I was following `firewall-allow-check`. Yes, a single drop rule is enough, have changed it.